controllers/flowlogspipeline/flp_common_objects.go

```diff
@@ -499,40 +484,58 @@ func (b *builder) addTransformStages(stage *config.PipelineBuilderStage) (*corev
 	}
 
 	// obtain encode_prometheus stage from metrics_definitions
-	promMetrics, dashboard, err := b.obtainMetricsConfiguration()
+	promMetrics, topPromMetrics, err := b.obtainMetricsConfiguration()
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	var dashboardConfigMap *corev1.ConfigMap
 	if len(promMetrics) > 0 {
 		// prometheus stage (encode) configuration
 		promEncode := api.PromEncode{
 			Prefix:  "netobserv_",
 			Metrics: promMetrics,
 		}
 		enrichedStage.EncodePrometheus("prometheus", promEncode)
-		dashboardConfigMap = b.makeMetricsDashboardConfigMap(dashboard)
 	}
 
-	b.addCustomExportStages(&enrichedStage)
-	return dashboardConfigMap, nil
-}
-
-func (b *builder) makeMetricsDashboardConfigMap(dashboard string) *corev1.ConfigMap {
-	configMap := corev1.ConfigMap{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      dashboardCMName,
-			Namespace: dashboardCMNamespace,
-			Labels: map[string]string{
-				dashboardCMAnnotation: "true",
+	if len(topPromMetrics) > 0 && b.info != nil && b.info.AvailableAPIs != nil && b.info.AvailableAPIs.HasSvcMonitor() {
+		topBytesStage := enrichedStage.ExtractTimebased("top_bytes", api.ExtractTimebased{
+			Rules: []api.TimebasedFilterRule{
+				{
+					Name:          "top_src_bytes",
+					IndexKeys:     []string{"SrcK8S_Name", "SrcK8S_Namespace", "SrcK8S_Type", "SrcK8S_HostName", "FlowDirection"},
+					OperationType: "avg",
+					OperationKey:  "Bytes",
+					TopK:          10,
+					Reversed:      false,
+					TimeInterval: api.Duration{
+						Duration: 60 * time.Second,
+					},
+				},
+				{
+					Name:          "top_dst_bytes",
+					IndexKeys:     []string{"DstK8S_Name", "DstK8S_Namespace", "DstK8S_Type", "DstK8S_HostName", "FlowDirection"},
+					OperationType: "avg",
+					OperationKey:  "Bytes",
+					TopK:          10,
+					Reversed:      false,
+					TimeInterval: api.Duration{
+						Duration: 60 * time.Second,
+					},
+				},
 			},
-		},
-		Data: map[string]string{
-			dashboardCMFile: dashboard,
-		},
+		})
+
+		// prometheus stage (encode) configuration for top metrics
+		topPromEncode := api.PromEncode{
+			Prefix:  "netobserv_",
+			Metrics: topPromMetrics,
+		}
+		topBytesStage.EncodePrometheus("top_prometheus", topPromEncode)
```

Comment:
The positive side of this approach is that it requires only two stages:
- gather top X src / dst bytes per Name / Namespace / Type / HostName 
- send to prometheus
The resources consumption are quite low on a small cluster.

However, since there is no filter on `Type` prior to this stage, the top X results may contains only pods or services in the dataset and generated graphs may be empty.

![image](https://github.com/netobserv/network-observability-operator/assets/91894519/67713f88-d14b-4312-8af5-4025b7095f35)
![image](https://github.com/netobserv/network-observability-operator/assets/91894519/dfba9187-1b85-4bb8-9b02-f1147ca68901)
![image](https://github.com/netobserv/network-observability-operator/assets/91894519/eea6b6ef-bd3b-447d-b6ec-51e768b93b62)

Ideally, we could improve the timebased stage to accept grouping. Else we'll need to add 4 parallel filter stages for Src Pod, Src Service, Dst Pod and Dst Service